src/Endpoints/CustomFields/Search/CustomFieldsSearchRequest.php

```diff
@@ -5,184 +5,69 @@
  */
-class CustomFieldsSearchRequest extends AbstractRequest
+class CustomFieldsSearchRequest extends AbstractSearchRequest
 {
-    public int $page;
-
-    public int $limit;
-
-    /**
-     * Comma separated list of properties to select. eg. "?select=id,name" If not provided, all fields are selected.
-     *
-     * Allowed values: "id", "name", "type"
-     */
-    public ?string $select = null;
-
     /**
      * Using this parameter, "customfield_options_url" property will be replaced by "customfield_options" contianing
      * expanded data. See examples below For more information see "/customfield-options" endpoint.
      *
      * Allowed values: "customfield_options"
      */
-    public ?string $expand = null;
-
-    /**
-     * Comma separated list of sorting keys from left side. Prepend "-" to any key for desc direction, eg.
-     * "?sort=type,-name"
-     *
-     * Allowed values: "id", "name", "type"
-     */
-    public ?string $sort = null;
-
-    /**
-     * Filters holder object
-     */
-    protected CustomFieldsSearchFilters $filters;
-
-    /**
-     * @param int|null $page  desired page
-     * @param int|null $limit Number of records on page. Maximum (default) allowed value is 500
-     */
-    public function __construct(Api $api, ?int $page = null, ?int $limit = null)
-    {
-        parent::__construct($api);
-
-        $this->page = $page ?? 1;
-        $this->limit = $limit ?? 100;
-        $this->filters = new CustomFieldsSearchFilters($this);
-    }
-
-    /**
-     * Current filters
-     */
-    public function filter(): CustomFieldsSearchFilters
+    public function expandBy(array $expand): self
     {
-        return $this->filters;
+        InvalidFormatException::checkAllowedValues($expand, CustomFieldDefinition::EXPAND_FIELDS);
+        return parent::expandBy($expand);
     }
 
-    /**
-     * Using this parameter, "customfield_options_url" property will be replaced by "customfield_options" contianing
-     * expanded data. See examples below For more information see "/customfield-options" endpoint.
-     *
-     * Allowed values: "customfield_options"
-     */
-    public function expandBy(string $expand): self
+    public function expandCustomFieldOptions(): self
     {
-        InvalidFormatException::checkInArray($expand, ['customfield_options']);
-        $this->expand = $expand;
-        return $this;
+        return $this->expandBy(['customfield_options']);
     }
 
     /**
      * Comma separated list of properties to select. eg. "?select=id,name" If not provided, all fields are selected.
      *
      * Allowed values: "id", "name", "type"
      */
-    public function select(?string $select): self
+    public function select(array $select): self
     {
-        $this->select = $select;
-        return $this;
+        InvalidFormatException::checkAllowedValues($select, CustomFieldDefinition::SELECT_FIELDS);
```

Comment:
Nechceš místo overridu (které máme nyní na více implementacích) dát abstraktní funkci která vynutí aby si vrátil seznam povolených select fieldů a v té abstraktní třídě volat ty checky?

Reply:
Takhle to bylo řešeno v původní implementaci tak jsme to zachoval, ale je fakt že ta abstraktní fce by asi byla lepší

Reply:
Můžeme se změnit v budoucnu :) díky!